src/features/applicationView/offerCard/editForm/EditForm.tsx

```diff
@@ -0,0 +1,134 @@
+} from './__generated__/CREATE_ORDER_LINE';
+import { CREATE_ORDER_LINE_MUTATION, DELETE_ORDER_LINE_MUTATION } from './mutation';
+import {
+  DELETE_ORDER_LINE,
+  DELETE_ORDER_LINEVariables as DELETE_ORDER_LINE_VARS,
+} from './__generated__/DELETE_ORDER_LINE';
+import { formatPrice } from '../../../../common/utils/format';
+import Button from '../../../../common/button/Button';
+import Text from '../../../../common/text/Text';
+
+export interface EditFormProps {
+  orderId: string;
+  selectedProducts: { productId: string; orderId: string }[];
+  handleCancel(): void;
+  handleSubmit(): void;
+}
+
+interface Values {
+  [productId: string]: boolean;
+}
+
+const EditForm: React.FC<EditFormProps> = ({ orderId, selectedProducts, handleCancel, handleSubmit }) => {
+  const { t, i18n } = useTranslation();
+  const { data, loading } = useQuery<ORDER_OPTIONAL_PRODUCTS>(ORDER_OPTIONAL_PRODUCTS_QUERY);
+  const [createOrderLine] = useMutation<CREATE_ORDER_LINE, CREATE_ORDER_LINE_VARS>(CREATE_ORDER_LINE_MUTATION);
+  const [deleteOrderLine] = useMutation<DELETE_ORDER_LINE, DELETE_ORDER_LINE_VARS>(DELETE_ORDER_LINE_MUTATION);
+
+  if (loading)
+    return (
+      <div className={styles.editForm}>
+        <LoadingSpinner isLoading={loading} />
+      </div>
+    );
+  if (!data?.additionalProducts) return <div className={styles.editForm}>{t('common.notification.noData')}</div>;
+
+  const initialValues = data.additionalProducts.edges.reduce<Values>((acc, edge) => {
+    if (!edge?.node) return acc;
+    const productNodeId = edge.node.id;
+
+    return {
+      ...acc,
+      [edge.node.id]: selectedProducts.find(({ productId }) => productId === productNodeId) ? true : false,
+    };
+  }, {});
+
+  const onSubmit = async (values: Values) => {
+    type CreateMutation = ReturnType<typeof createOrderLine>;
+    type DeleteMutation = ReturnType<typeof deleteOrderLine>;
+
+    const createMutations: CreateMutation[] = [];
+    const deleteMutations: DeleteMutation[] = [];
+
+    Object.keys(initialValues).forEach((key) => {
+      if (initialValues[key] === values[key]) return;
+      if (values[key] === true)
+        return createMutations.concat([createOrderLine({ variables: { input: { orderId, productId: key } } })]);
+      if (values[key] === false) {
+        const product = selectedProducts.find(({ productId }) => productId === key);
+        return product && deleteMutations.concat([deleteOrderLine({ variables: { input: { id: product.orderId } } })]);
+      }
+    });
+
+    await Promise.all<CreateMutation | DeleteMutation>([...createMutations, ...deleteMutations]);
+
+    handleSubmit();
+  };
+
+  return (
+    <Formik
+      initialValues={initialValues}
+      onSubmit={(values) => onSubmit(values)}
+      validateOnBlur={false}
+      validateOnChange={false}
+    >
+      {({ values, errors, handleChange, handleSubmit }) => {
+        type Product = { id: string; label: string; price: number };
```

Comment:
Should this type be outside of the return block?

Reply:
Moved it up